src/code/z_play.c

```diff
@@ -403,7 +403,7 @@ void Play_Init(GameState* thisx) {
                  (s32)(zAllocAligned + zAllocSize) - (s32)(zAllocAligned - zAlloc));
 
     Fault_AddClient(&D_801614B8, ZeldaArena_Display, NULL, NULL);
-    func_800304DC(this, &this->actorCtx, this->linkActorEntry);
+    func_800304DC(this, &this->actorCtx, this->playerEntry);
```

Comment:
Looking at `func_800304DC()`, should the last arg in that function be named `playerEntry` as well?

Reply:
its a tough question, cause nothing about it is player specific. it just that play passes player to it.
looking at it from the actor systems point of view, it spawns player first so that it can pass it to `func_8002C0C0` which initlizes target context stuff. The target context stuff then goes on to pass that actor instance around to other non-player specific functions.

play also wants player to be spawned first because in Play_Init it will then do some camera setup that needs his instance.
So, if you were to pass anything but player there, things would technically break outside of this function call. But in the scope of this function call, all that really matters is that its the first actor to get spawned as the whole context is initialized.

I dont really have an opinion yet, its kinda a messy situation lol. more thoughts welcome

Reply:
The initialization of target context via `func_8002C0C0` happens internally within `func_800304DC`, so that does make it seem player-dependent.

So internally within `func_800304DC`:
    1) `bzero(actorCtx, sizeof(*actorCtx));`
        - This sets all of actorContext to zero, including `actorLists`. So if player was spawned first before this function, that zeroing out resets it
    2) `Actor_SpawnEntry(actorCtx, actorEntry, play);`
        - This spawn the first entry to actorLists
    3) `func_8002C0C0(&actorCtx->targetCtx, actorCtx->actorLists[ACTORCAT_PLAYER].head, play);`
        - It's expecting player to already be here, even though it was zeroed out a few lines earlier

So I think internally, it makes sense that it should be `playerEntry`, but yeah open to more thoughts

Reply:
yea thats fair enough, ill change it. i also will take the opportunity to just name `func_800304DC` as `Actor_InitContext`, the comment has been above it for years

Reply:
Would also be nice to change:
`bzero(actorCtx, sizeof(*actorCtx));` -> `bzero(actorCtx, sizeof(ActorContext));`
since I think that's what we prefer, but could always do it later.

Reply:
I think `*actorCtx` is slightly better since it avoids duplicating the type (which is generally good practice), but it's very minor since this is C and the type is everywhere anyway

Reply:
we are at the very least pretty inconsistent about it for just bzero calls, probably moreso in general. So its something we'll need to hash out at a larger scale.
Ive already changed it, and prefer it a bit more personally, but yeah I do understand how it can be seen as better practice to do it the other way

Reply:
Yeah it's definitely not consistent right now, so it's fine either way